src/v/bytes/details/io_fragment.h

```diff
@@ -90,6 +90,7 @@ class io_fragment {
     void trim(size_t len) { _used_bytes = std::min(len, _used_bytes); }
     void trim_front(size_t pos) {
         // required by input_stream<char> converter
+        _used_bytes -= pos;
```

Comment:
this is good, but wondering if we should `vassert()` here. this is ok. 

only question is should this be a signed int. (separate commit) cc: @travisdowns